docs/content/basic/custom-steps.md

```diff
@@ -0,0 +1,88 @@
+---
+title: Listening and responding to custom steps
+lang: en
+slug: /concepts/custom-steps
+---
+
+Your app can use the `function()` method to listen to incoming [custom step requests](https://api.slack.com/automation/functions/custom-bolt). Custom steps are used in Workflow Builder to build workflows. The method requires a step `callback_id` of type `str`. This `callback_id` must also be defined in your [Function](https://api.slack.com/concepts/manifests#functions) definition. Custom steps must be finalized using the `complete()` or `fail()` listener arguments to notify Slack that your app has processed the request.
+
+* `complete()` requires **one** argument: `outputs` of type `dict`. It completes your custom step **successfully** and provides a dictionary containing the outputs of your custom step as per its definition.
+* `fail()` requires **one** argument: `error` of type `str`. It completes your custom step **unsuccessfully** and provides a message containing information regarding why your custom step failed.
+
+You can reference your custom step's inputs using the `inputs` listener argument of type `dict`.
+
+Refer to [the module document](https://slack.dev/bolt-python/api-docs/slack_bolt/kwargs_injection/args.html) to learn about the available listener arguments.
+
+```python
+# This sample custom step formats an input and outputs it
+@app.function("sample_custom_step")
+def sample_step_callback(inputs: dict, ack: Ack, fail: Fail, complete: Complete):
+    try:
+        ack()
```

Comment:
```suggestion
def sample_step_callback(inputs: dict, fail: Fail, complete: Complete):
    try:
```

Calling `ack` seems to be valid here but I don't think it's necessary 🤔  Please correct me if this is wrong though!

Reply:
I prefer to keep it in the docs, there are a number of issues opened related to calling `ack` before executing custom logic

Keeping it in hopes that it will help developers avoid a bug 🙏